blockly-core/core/ui/block.js

```diff
@@ -1796,7 +1809,7 @@ Blockly.Block.prototype.setNextStatement = function(hasNext, opt_check) {
     this.nextConnection.dispose();
     this.nextConnection = null;
   }
-  if (hasNext) {
+  if (hasNext && !this.nextConnectionDisabled_) {
```

Comment:
What does this buy us? it looks like individual block definitions call setNextStatement(true) before `domToBlock` sets `nextConnectionDisabled_` .